server/src/main/java/org/elasticsearch/index/search/QueryParserHelper.java

```diff
@@ -177,18 +132,20 @@ public static Mapper getFieldMapper(MapperService mapperService, String field) {
                 continue;
             }
 
-            // Ignore fields that are not in the allowed mapper types. Some
-            // types do not support term queries, and thus we cannot generate
-            // a special query for them.
-            String mappingType = fieldType.typeName();
-            if (acceptAllTypes == false && ALLOWED_QUERY_MAPPER_TYPES.contains(mappingType) == false) {
+            if (acceptMetadataField == false && fieldType.name().startsWith("_")) {
+                // Ignore metadata fields
                 continue;
             }
 
-            // Ignore metadata fields.
-            Mapper mapper = getFieldMapper(context.getMapperService(), fieldName);
-            if (acceptMetadataField == false && mapper instanceof MetadataFieldMapper) {
-                continue;
+            if (acceptAllTypes == false) {
+                try {
+                    fieldType.termQuery("", context);
+                } catch (QueryShardException |UnsupportedOperationException e) {
+                    // field type is never searchable with term queries (eg. geo point): ignore
+                    continue;
+                } catch (RuntimeException e) {
+                    // other exceptions are parsing errors or not indexed fields: keep
```

Comment:
Should we not be throwing the exception here and reporting the problem back to the user?

Reply:
No, this case is for field mappers that need a special input (numbers, boolean, ...) so this is consistent with the current behavior. The goal here is to select fields that can handle a `term` query so we silently ignore those that can't. Previously this selection was based on an hardcoded list of field types.

Reply:
can we narrow down the exception type here, eg. IAE?

Reply:
True thanks, it should be IAE only. I pushed the change.